### ISSUE TITLE
Fix image size in PDF

### ISSUE THREAD

**Issue by SankethBK** [SankethBK/diaryvault#226]
Currently when a note containing image is exported into PDF, the image dimensions are retained from uploaded image and it fills up 2-3 pages in PDF. 

For example this is the note

<img src="https://github.com/user-attachments/assets/cb00b258-f076-433b-aca6-8eadd99705b2" width="300" />

Pdf will be something like this

<img src="https://github.com/user-attachments/assets/f584e1aa-72ac-47a7-ba6b-0e4e6f3839dd" width="300" />

PDF is generated from `exportNotesToPDF` function which eventually calls `preprocessDeltaForPDFExport` function in `lib/features/notes/data/repositories/export_notes_repository.dart`. This function first converts delta format to HTML and then to PDF. We can access each delta element and its attributes inside this function. We need to edit the `img` tags generated in HTML to and add suitable `height`, `width` attributes so that the image doesn't span over multiple pages in PDF.

Relevant widgets
```
- packages/flutter_html_to_pdf/lib/flutter_html_to_pdf.dart
```

**Comment by xkaper001** [comment]
Hi @SankethBK ,

I noticed that this issue is unassigned, and I'd love to work on it! I'm familiar with the technology used in this project and I believe I can help resolve it.

Could you please assign this issue to me?

Thank you!

**Comment by SankethBK** [comment]
Hi @xkaper001 , please go ahead

**Comment by vaishnavi-2301** [comment]
Hey @SankethBK, I took a look at the issue, and I believe I resolved it, kindly take a look at the pr. https://github.com/SankethBK/diaryvault/pull/234, and let me know if it needs to be modified.

**Comment by xkaper001** [comment]
> Hey @xkaper001,
> 
> I understand your concern, but this is an open-source project, which means anyone can contribute. I had already looked into this issue, worked on a solution, and submitted a PR before realizing it was assigned to you. You're welcome to take a look at my PR, and we can collaborate or build on it from there. Open-source contributions are meant to be collaborative, so I'm happy to discuss any changes or improvements!
> 
> 

Yeah exactly an open source project means anyone can contribute. but think logically what exactly is the point of multiple people working on the same issue independently when only one's PR will get merged. That's not how open source works.

Again I am not saying your PR is wrong. but the approach is. Next time keep in mind to ask the maintainers to get the issue assigned to yourself or ask to collaborate on the issue if someone's already working on it. That's how open source works.

**Comment by SankethBK** [comment]
Hey @xkaper001 and @vaishnavi-2301 , I truly appreciate the time and effort each of you has put into working on this issue. 

@vaishnavi-2301  Thank you for working on this issue. I understand that since this is a small issue, you were probably experimenting with a proof of POC and arrived at a solution. Having issues assigned before starting work is helpful to avoid overlap and ensure we use everyone's time effectively. 

@xkaper00 If you haven't started working on this, you can review @vaishnavi-2301 's PR and collaborate to add any improvements if necessary. If you already have a working fix, feel free to submit your own PR. I'll review both.  

I'll also be adding more issues today, so there will be plenty of opportunities for everyone to contribute.

**Comment by xkaper001** [comment]
Thanks @SankethBK, tho I had started working on this one I'll prefer working on newer issues since this was already fixed.
